### ISSUE TITLE
Train & Export CNN-based Classifier

### ISSUE THREAD

**Issue by Anushka26Ml** [Anushka26Ml/Marchin#4]
**As a** machine learning engineer
 **I need**   to train a CNN-based classifier (or an ensemble using CNN embeddings) and export model weights 
 **So that** the system can classify eDNA sequences into known taxa and the model is ready for deployment.
   
 ### Details and Assumptions
 * Model architecture: CNN-based classifier built in TensorFlow (optionally combined with a lightweight classifier head or ensemble).
 * Export format: ONNX for inference portability; keep TensorFlow checkpoints for retraining.
 * Evaluation uses held-out validation set and standard metrics (accuracy, precision, recall).
   
 ### Acceptance Criteria  
   
 ```gherkin
Given CNN embeddings or preprocessed sequence inputs,
When training begins,
Then model training metrics (loss, accuracy) must be logged.

Given training completes,
When evaluation runs on the validation set,
Then classification metrics (accuracy, precision, recall) must be reported and meet the defined baseline.

Given the model passes validation,
When exporting for deployment,
Then weights must be saved in ONNX format and TensorFlow checkpoints must be archived.
 ```
